numpy/_core/src/multiarray/abstractdtypes.c

```diff
@@ -85,17 +85,29 @@ discover_descriptor_from_pycomplex(
 NPY_NO_EXPORT int
 initialize_and_map_pytypes_to_dtypes()
 {
-    ((PyTypeObject *)&PyArray_PyIntAbstractDType)->tp_base = &PyArrayDescr_Type;
+    if (PyType_Ready((PyTypeObject *)&PyArray_IntAbstractDType) < 0) {
+        return -1;
+    }
+    if (PyType_Ready((PyTypeObject *)&PyArray_FloatAbstractDType) < 0) {
+        return -1;
+    }
+    if (PyType_Ready((PyTypeObject *)&PyArray_ComplexAbstractDType) < 0) {
+        return -1;
+    }
+    ((PyTypeObject *)&PyArray_PyIntAbstractDType)->tp_base =
```

Comment:
Note to self: need to add a note that these delayed assignments are necessary for Windows (if one does it in the struct, one gets `error C2099: initializer is not a constant"`).